libafl/src/stages/tracing.rs

```diff
@@ -100,6 +103,148 @@ impl<EM, TE, Z> TracingStage<EM, TE, Z> {
     }
 }
 
+/// Trace with tainted input
+#[derive(Clone, Debug)]
+pub struct TaintTracingStage<EM, TE, Z> {
+    tracer_executor: TE,
+    cmplog_observer_name: Option<String>,
+    #[allow(clippy::type_complexity)]
+    phantom: PhantomData<(EM, TE, Z)>,
+}
+
+impl<EM, TE, Z> UsesState for TaintTracingStage<EM, TE, Z>
+where
+    TE: UsesState,
+{
+    type State = TE::State;
+}
+
+impl<E, EM, TE, Z> Stage<E, EM, Z> for TaintTracingStage<EM, TE, Z>
```

Comment:
Isn't this doing taint tracing right? We should find a better name

Reply:
maybe DoubleTracingStage?